lib/src/component/abstract_transition.dart

```diff
@@ -100,37 +100,95 @@ abstract class AbstractTransitionComponent<T extends AbstractTransitionProps,
   );
 
   @override
-  get initialState => (newState()
-    ..transitionPhase = this.initiallyShown ? TransitionPhase.SHOWN : TransitionPhase.HIDDEN
-  );
+  get initialState {
+    return (newState()
+      ..transitionPhase = this.initiallyShown ? TransitionPhase.SHOWN : TransitionPhase.HIDDEN
+    );
+  }
+
+  @mustCallSuper
+  @override
+  Map getDerivedStateFromProps(Map nextProps, Map prevState) {
+    final tNextProps = typedPropsFactory(nextProps);
+    if (tNextProps.shown == null) return prevState;
+
+    final tPrevState = typedStateFactory(prevState);
+    if (tNextProps.shown && _isOrWillBeHidden(tPrevState)) {
+      return _handleShow(tNextProps, tPrevState, this);
+    } else if (!tNextProps.shown && _isOrWillBeShown(tPrevState)) {
+      return _handleHide(tNextProps, tPrevState, this);
+    }
+
+    return prevState;
+  }
+
+  static bool _isControlled(AbstractTransitionProps tProps) => tProps.shown != null;
 
   /// Stream for listening to `transitionend` events on the [AbstractTransitionComponent].
   StreamSubscription _endTransitionSubscription;
 
   /// Whether the [AbstractTransitionComponent] should be visible initially when mounted.
-  bool get initiallyShown;
+  bool get initiallyShown => props.shown ?? false;
 
   /// Returns the DOM node that will transition.
   Element getTransitionDomNode();
 
+  /// Returns the value of [TransitionPropsMixin.transitionCount], falling back to `null` for backwards compatibility.
+  static int _getTransitionCount(AbstractTransitionProps tProps) => tProps?.transitionCount ?? 1;
+
   /// Whether transitions are enabled for this component.
-  bool get hasTransition => true;
+  ///
+  /// Will not work when the component is controlled using [TransitionPropsMixin.shown].
+  /// Use [TransitionPropsMixin.transitionCount] instead.
+  ///
+  /// __Deprecated.__ Use [TransitionPropsMixin.transitionCount] instead.
+  @Deprecated('Use `props.transitionCount` instead.')
+  bool get hasTransition => _hasTransition(props);
```

Comment:
Same question as below; what's the migration path? https://sourcegraph.wk-dev.wdesk.org/search?q=get%5CshasTransition+repo:%5Egithub%5C.com/Workiva/web_skin_dart%24+&patternType=regexp#1